### PR TITLE
Refactor FXIOS-9661 Change nimbus feature flag for address autofill phase 2 [ca, us]

### PR DESCRIPTION
**Pull request by razvanlitianu**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9661)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21265)

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift [hunk 1]
```diff
@@ -2503,7 +2503,7 @@ class BrowserViewController: UIViewController,
     }
 
     private func addressAutofillNimbusFeatureFlag() -> Bool {
```
Comment (nbhasin2) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:2505]:
```suggestion
      private func addressAutofillLocaleFeatureFlag() -> Bool {
```

**Comment by issammani** [comment]
@nbhasin2 this looks good already. I think minor detail is maybe we want one method to do this check: `["CA", "US"].contains(Locale.current.regionCode ?? ""`. I see pocket has something [similar](https://github.com/mozilla-mobile/firefox-ios/blob/8a36d9a2b0699f1183ebe167bb1f1a2d8d1b8307/firefox-ios/Providers/Pocket/PocketProvider.swift#L119-L121). What do you think about adding a static method `isAdressAvailable`  to `AddressProvider` as an extension. I can make the change if this makes sense 😄

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift (hunk 1 above)
Comment (nbhasin2) [reply firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift:2505]: cc @issammani 

> @nbhasin2 this looks good already. I think minor detail is maybe we want one method to do this check: `["CA", "US"].contains(Locale.current.regionCode ?? ""`. I see pocket has something [similar](https://github.com/mozilla-mobile/firefox-ios/blob/8a36d9a2b0699f1183ebe167bb1f1a2d8d1b8307/firefox-ios/Providers/Pocket/PocketProvider.swift#L119-L121). What do you think about adding a static method `isAdressAvailable` to `AddressProvider` as an extension. I can make the change if this makes sense 😄

Based on your comment we can add it to the address provider

**Review by nbhasin2 — APPROVED** [review]

File: firefox-ios/Client/Frontend/Settings/SyncContentSettingsViewController.swift [hunk 2]
```diff
@@ -265,9 +265,7 @@ class SyncContentSettingsViewController: SettingsTableViewController, FeatureFla
             engineSectionChildren.append(creditCards)
         }
 
-        if featureFlags.isFeatureEnabled(
-            .addressAutofill,
-            checking: .buildOnly) {
+        if ["CA", "US"].contains(Locale.current.regionCode ?? "") {
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Settings/SyncContentSettingsViewController.swift:268]: why not replace `AddressLocaleFeatureValidator.isValidRegion()` like in other areas?

File: firefox-ios/Client/Frontend/Autofill/Address/AddressProvider.swift [hunk 3]
```diff
@@ -14,3 +14,14 @@ protocol AddressProvider {
...
+
+struct AddressLocaleFeatureValidator {
+    static let supportedRegions = ["CA", "US"]
+
+    static func isValidRegion() -> Bool {
+        guard let regionCode = Locale.current.regionCode else {
+            return false
+        }
+        return supportedRegions.contains(regionCode)
+    }
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Autofill/Address/AddressProvider.swift:26]: nit: should we move this to a new file and maybe test? 🤣

**Review by cyndichin — APPROVED** [review]
thanks for addressing so quickly, looks good!

File: firefox-ios/Client/Frontend/Autofill/Address/AddressProvider.swift (hunk 3 above)
Comment (nbhasin2) [reply firefox-ios/Client/Frontend/Autofill/Address/AddressProvider.swift:26]: File is fine, but let me add tests. I can add them today although it feels wrong to push it so close to the deadline to we will do it for v129.1 

Going to first add tests and only then merge it in.

**Review by MattLichtenstein — APPROVED** [review]
✅

File: firefox-ios/Client/Frontend/Autofill/Address/AddressProvider.swift (hunk 3 above)
Comment (nbhasin2) [reply firefox-ios/Client/Frontend/Autofill/Address/AddressProvider.swift]: cc @MattLichtenstein @cyndichin added some tests here

**Review by MattLichtenstein — APPROVED** [review]
👌

**Review by cyndichin — APPROVED** [review]
looks good, thank you for making the updates + adding tests! 🔥

**Comment by nbhasin2** [comment]
@mergify backport release/v129

**Pull request merged**